File: react/react-demo/src/demo/images/_gen_avatar.js
const { createCanvas } = require('canvas');
const fs = require('fs');
const path = require('path');

const size = 200;
const canvas = createCanvas(size, size);
const ctx = canvas.getContext('2d');

// Blue circle background
ctx.fillStyle = '#00aeec';
ctx.fillRect(0, 0, size, size);

// Draw "bo" text
ctx.fillStyle = '#ffffff';
ctx.font = 'bold 80px Arial';
ctx.textAlign = 'center';
ctx.textBaseline = 'middle';
ctx.fillText('bo', size / 2, size / 2);

const outPath = path.join(__dirname, 'bozai.png');
const buf = canvas.toBuffer('image/png');
fs.writeFileSync(outPath, buf);
console.log('Saved:', outPath);

File: react/react-demo/src/demo/images/_gen_avatar.py
from PIL import Image, ImageDraw, ImageFont
import os

size = 200
img = Image.new('RGBA', (size, size), (0, 174, 236, 255))
draw = ImageDraw.Draw(img)

# Draw a simple "bo" text avatar
try:
    font = ImageFont.truetype("arial.ttf", 80)
except:
    font = ImageFont.load_default()

text = "bo"
bbox = draw.textbbox((0, 0), text, font=font)
text_w = bbox[2] - bbox[0]
text_h = bbox[3] - bbox[1]
draw.text(((size - text_w) / 2, (size - text_h) / 2 - 5), text, fill="white", font=font)

output = os.path.join(os.path.dirname(__file__), "bozai.png")
img.save(output, "PNG")
print(f"Saved: {output}")
